### PR TITLE
Add tests for rotation averaging edge cases

### PR DESCRIPTION
**Pull request by ahojnnes**
## Summary
- Add 16 new tests for `estimators/rotation_averaging.cc` targeting previously uncovered code paths
- Existing 5 tests preserved unchanged; new tests cover option variations, error handling, and alternative solver paths

## New test cases
| Test | Code path covered |
|------|------------------|
| `SkipInitialization` | `skip_initialization=true` bypasses spanning tree init |
| `HalfNormWeightType` | `HALF_NORM` weight in IRLS solver |
| `DeterministicRandomSeed` | `random_seed >= 0` deterministic mode |
| `DisableRotationErrorFiltering` | `max_rotation_error_deg <= 0` skips filtering |
| `EmptyPoseGraphReturnsFalse` | Empty LCC returns false (known rig path) |
| `GravityWithUnknownRigSensorsReturnsFalse` | `AllSensorsFromRigKnown` returning false |
| `FilterUnregistered` | `filter_unregistered=true` refinement pass |
| `StratifiedGravityAllPairsHaveGravity` | >95% gravity pairs skips subset solve |
| `GravityWithoutStratified` | `use_stratified=false` with gravity |
| `GravityEnabledButNoPriors` | `UseGravity` returns false when no priors exist |
| `InitializeRigRotationsFromImagesMultiCamera` | Multi-camera rig init with unknown cam_from_rig |
| `InitializeRigRotationsFromImagesEmpty` | Empty cams_from_world input |
| `RotationErrorFilteringDeregistersOutliers` | Tight threshold triggers deregistration |
| `FilterEdgesInvalidatesOutlierPairs` | Edge invalidation by rotation error |
| `WithNoiseAndUnknownRig` | Full expanded reconstruction path with noise |
| `EmptyPoseGraphWithUnknownRigReturnsFalse` | Empty LCC in expanded reconstruction branch |

## Test plan
```
cd build && ./src/colmap/estimators/glomap_rotation_averaging_test
[==========] Running 21 tests from 1 test suite.
[  PASSED  ] 21 tests.
```
All 21 tests pass (5 existing + 16 new) in ~530ms total.

### CODE REVIEW

File: src/colmap/estimators/rotation_averaging_test.cc [hunk 1]
```diff
@@ -287,5 +287,681 @@ TEST(RotationAveraging, WithNoiseAndOutliersWithNonTrivialKnownRigs) {
...
+
+  RotationEstimatorOptions options = CreateRATestOptions();
+  options.skip_initialization = true;
+
+  Reconstruction reconstruction_copy = reconstruction;
+  // Should still succeed even without spanning tree initialization,
+  // since the solver can work from zero-initialized rotations.
+  EXPECT_TRUE(RunRotationAveraging(
+      options, pose_graph, reconstruction_copy, pose_priors));
+}
```
[the remark this answers is not in the document]
Comment (ahojnnes) [reply src/colmap/estimators/rotation_averaging_test.cc]: Acknowledged. Keeping tests self-contained for now - each test documents its own setup, making it easier to understand in isolation. Happy to refactor into a fixture if a reviewer prefers.

File: src/colmap/estimators/CMakeLists.txt [hunk 2]
```diff
@@ -116,7 +116,7 @@ COLMAP_ADD_TEST(
 )
 
 COLMAP_ADD_TEST(
-    NAME glomap_rotation_averaging_test
+    NAME rotation_averaging_test
```
Comment (B1ueber2y) [inline src/colmap/estimators/CMakeLists.txt:119]: Nice catch. This naming was not fixed in the migration. There are some other similar cases in this codebase.
